### DISCUSSION TITLE
Request a lighter js library without manipulating localstorage

### DISCUSSION THREAD

**Discussion by hallee9000** [supabase/supabase#2413] [Feature Requests]
Hi there, I'm working on a Figma plugin and integrating supabase-js in it. Figma plugin is running in [a iframe sandbox where localstorage is disabled](https://www.figma.com/plugin-docs/how-plugins-run/), so I got this error.

![image](https://user-images.githubusercontent.com/12868055/126025462-55febd7a-fc22-4ef8-9317-56c9bbe3391f.png)

I wonder if supabase can provide a lighter library without manipulating localstorage, then I can handle these stuff with [Figma plugin client storage](https://www.figma.com/plugin-docs/api/figma-clientStorage) by myself.

Thanks.

**Comment by inian** [comment]
What part of the library are you using @leadream? supabase-js is a thin wrapper for [these libraries](https://github.com/supabase/supabase-js/blob/2ffa2d9ed60cd469e40c793e67dd56d59a13c14f/package.json#L39-L42)

I think only gotrue-js uses localstorage. If for example, you just need to interact with our rest API, can you try using postgrest-js directly which shouldn't depend on localstorage.

**Reply by hallee9000** [reply]
Oh thanks @inian , I use the entire `supabase-js`. I just figured it out by using `postgrest-js`. It's not easy, because to achieve it I had to read some source code. I guess the docs for `postgrest-js` should be more clear, or add some examples.

BTW I still have two questions here
- How to implement `auth.login` by `postgrest-js`
- How to upload files by `postgrest-js`

Post code here in case someone has the same issue.

```ts
// postgrest.ts
import { PostgrestClient } from '@supabase/postgrest-js'
const SUPABASE_ANON_KEY = process.env.SUPABASE_ANON_KEY
const SUPABASE_URL = process.env.SUPABASE_URL

function getAuthHeaders (user) {
  const headers: { [key: string]: string } = {}
  const authBearer = user?.access_token ?? SUPABASE_ANON_KEY
  headers['apikey'] = SUPABASE_ANON_KEY
  headers['Authorization'] = `Bearer ${authBearer}`
  return headers
}

export async function initSupabase (user) {
  const REST_URL = `${SUPABASE_URL}/rest/v1`
  const postgrest = new PostgrestClient(REST_URL, {
    schema: 'public',
    headers: getAuthHeaders(user)
  })
  return postgrest
}
```

```ts
// another file
// `user` is from somewhere you store it, in figma plugin I store it in client storage
const postgrest = initSupabase(user)
const { data, error } = await postgrest
  .from('cities')
  .select()
```
